src/backend/distributed/operations/citus_tools.c

```diff
@@ -60,6 +63,37 @@ static Tuplestorestate * CreateTupleStore(TupleDesc tupleDescriptor,
 										  StringInfo *resultArray, int commandCount);
 
 
+/*
+ * citus_check_connection_to_node sends a simple query from a worker node to another
+ * node, and returns success status.
+ */
+Datum
+citus_check_connection_to_node(PG_FUNCTION_ARGS)
+{
+	char *nodeName = PG_GETARG_TEXT_TO_CSTRING(0);
+	int32 nodePort = PG_GETARG_INT32(1);
+
+	bool success = CheckConnectionToNode(nodeName, nodePort);
```

Comment:
Are we ok to connect to the node itself if given nodeName and nodePort is same with node's own info? And can we check opening a connection to node itself works as expected on Marlin. (I don't expect a problem, but just want to make sure)

Reply:
Ok, as I checked the tests we are ok to connect to node itself. Can we only double-check it on the Marlin?

Reply:
> Can we only double-check it on the Marlin?

I guess that does not impact this PR? 

And, if the nodes cannot connect back to itself, several features can be broken, like repartition joins or MX. So, I'd expect it to work :)